crypto/x509/v3_alt.c

```diff
@@ -128,7 +128,7 @@ STACK_OF(CONF_VALUE) *i2v_GENERAL_NAME(X509V3_EXT_METHOD *method,
                 BIO_snprintf(othername, sizeof(othername), "othername: %s:",
                              oline);
             else
-                strncpy(othername, "othername:", sizeof(othername));
+                strncpy(othername, "othername:", sizeof(othername) - 1);
```

Comment:
Since it's: char othername[300], this doesn't seem to fix anything. But if you want to fix something, you would need to make sure it's actually NUL terminated.

Reply:
`strncpy` [should](https://en.cppreference.com/w/c/string/byte/strncpy)

> Zeroes out the rest of the destination buffer

so there shouldn't be a need to explicitly NUL-terminate it.

Or am I missing your point about it?

---

I can see how there is no real risk of exceeding 300 when copying the shorter `"othername:"`, but conceptually shouldn't we just avoid the anti-pattern `strncpy(dst_buf, src_buf, sizeof(dst_buf))` as error prone?

Reply:
The point is, if you use sizeof()-1, you don't actually zero terminate it if your source string was >= sizeof()-1. You should make sure it's terminated by writing the '\0' to [sizeof()-1], because you didn't write anything to that. Or you could just use strlcpy().

Reply:
Right! Thanks for the feedback!

Reply:
In the latest fixup commit I just replaced `strncpy()` with `OPENSSL_strlcpy()`, I am not checking the returned length as it seems like we were already ok with truncating it here.